integration/docker/entrypoint.sh

```diff
@@ -113,41 +94,124 @@ function mountAlluxioRootFSWithFuseOption {
   tail -f /opt/alluxio/logs/fuse.log
 }
 
-case ${service,,} in
-  master)
-    formatMasterIfSpecified
-    integration/docker/bin/alluxio-job-master.sh &
-    integration/docker/bin/alluxio-master.sh &
-    wait -n
-    ;;
-  master-only)
-    formatMasterIfSpecified
-    integration/docker/bin/alluxio-master.sh
-    ;;
-  job-master)
-    integration/docker/bin/alluxio-job-master.sh
-    ;;
-  worker)
-    formatWorkerIfSpecified
-    integration/docker/bin/alluxio-job-worker.sh &
-    integration/docker/bin/alluxio-worker.sh &
-    wait -n
-    ;;
-  worker-only)
-    formatWorkerIfSpecified
-    integration/docker/bin/alluxio-worker.sh
-    ;;
-  job-worker)
-    integration/docker/bin/alluxio-job-worker.sh
-    ;;
-  proxy)
-    integration/docker/bin/alluxio-proxy.sh
-    ;;
-  fuse)
-    mountAlluxioRootFSWithFuseOption
-    ;;
-  *)
+# Sends a signal to each of the running background processes
+#
+# Args:
+#     1: the signal to send
+function forward_signal {
+  local signal="${1}"
+  # background jobs don't respond to SIGINT (2)
+  # Change to SIGHUP (1)
+  if [ "${signal}" -eq "2" ]; then
+    signal="1"
+  fi
+
+  local procs="$(jobs -p)"
+  echo -e "Forwarding signal ${signal} to processes:\n${procs}"
+  while read -r proc; do
+    if [ -n "${proc}" ]; then
+      kill -${signal} "${proc}"
+    fi
+  done <<< "${procs}"
+  # This function may take over execution thread from the "main" function.
+  # Wait if the processes are still up. Additional signals of the same type as this
+  # will not be able to be processed
+  wait
+}
+
+# Sets up traps on all signals [1, 31]
+#
+# Notes about trapping some signals
+# - SIGINT (2): Background process (started with &) ignore SIGINT. As a workaround to still
+#               terminate processes when SIGINT is passed, convert the signal sent to the
+#               processes to be something other than SIGINT
+# - SIGKILL (9): Cannot be trapped. It will directly kill the bash parent shell, the child
+#                processes will continue to live
+function setup_signals {
+  for i in {1..31}; do
+    trap "forward_signal ${i}" ${i}
+  done
+
+  # If the script exits for any reason without a signal, forward a SIGHUP to the children
+  trap "forward_signal 1" EXIT
+}
+
+function main {
+  if [[ "$#" -lt 1 ]]; then
+    printUsage
+    exit 1
+  fi
+
+  local service="$1"
+  local options="$2"
```

Comment:
Will local `options` work for other functions?

Reply:
Good catch! I will move options back to a global variable